### PR TITLE
Add sync db CLI

### PR DESCRIPTION
**Pull request by yuzhu**

### CODE REVIEW

**Comment by yuzhu** [comment]
@gpang Add sync CLI

**Comment by yuzhu** [comment]
I will add integration test once the test PR is merged

**Comment by yuzhu** [comment]
jenkins, test this please

File: table/server/master/src/main/java/alluxio/master/table/Database.java [hunk 1]
```diff
@@ -183,11 +185,13 @@ public void sync(JournalContext context) throws IOException {
         Journal.JournalEntry entry = Journal.JournalEntry.newBuilder().setAddTable(addTableEntry)
             .build();
         applyAndJournal(context, entry);
+        returnVal = true;
       } else {
         // sync metadata from udb
         // TODO(gpang): implement
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/Database.java]: we will need to discover new partitions in this part.

Comment (yuzhu) [reply table/server/master/src/main/java/alluxio/master/table/Database.java]: yeah, it is.. it goes through the Udb and if the table is not found in the correct table collection, it adds the table.

Comment (yuzhu) [reply table/server/master/src/main/java/alluxio/master/table/Database.java]: oops, you mean new partitions..

**Comment by yuzhu** [comment]
@gpang PTAL, comments addressed

File: table/server/master/src/main/java/alluxio/master/table/Table.java [hunk 2]
```diff
@@ -87,6 +90,50 @@ public void sync(UdbTable udbTable) {
...
+    return Objects.equals(partitionCols, mPartitionScheme.getPartitionCols());
+  }
+
+  /**
+   * incremental sync the table with a udbtable.
+   *
+   * @param udbTable udb table to be synced
+   * @return true if the table changed
+   */
+  public boolean incrementalSync(UdbTable udbTable) {
```
Comment (gpang) [inline table/server/master/src/main/java/alluxio/master/table/Table.java]: Is there not a way for just to have a single `sync()` method which handles all the sync cases?

**Comment by yuzhu** [comment]
@gpang comments addressed, PTAL

**Comment by yuzhu** [comment]
alluxio-bot, merge this please

**Comment by yuzhu** [comment]
alluxio-bot, cherry-pick this to branch-2.1 please

**Pull request merged**